src/ensembl/src/content/app/browser/browser-region-editor/BrowserRegionEditor.tsx

```diff
@@ -0,0 +1,340 @@
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+
+import Select from 'src/shared/components/select/Select';
+import Input from 'src/shared/components/input/Input';
+import Tooltip, { Position } from 'src/shared/components/tooltip/Tooltip';
+
+import { ChrLocation, RegionValidationResponse } from '../browserState';
+import { RootState } from 'src/store';
+import {
+  getRegionEditorActive,
+  getRegionFieldActive,
+  getBrowserActiveGenomeId,
+  getChrLocation,
+  getRegionValidationInfo,
+  getRegionValidationLoadingStatus
+} from '../browserSelectors';
+import { getGenomeKaryotypes } from 'src/genome/genomeSelectors';
+import {
+  changeBrowserLocation,
+  toggleRegionEditorActive,
+  resetRegionValidation,
+  validateRegion
+} from '../browserActions';
+import { GenomeKaryotype } from 'src/genome/genomeTypes';
+import { LoadingState } from 'src/shared/types/loading-state';
+
+import {
+  getCommaSeparatedNumber,
+  getNumberWithoutCommas
+} from 'src/shared/helpers/numberFormatter';
+import {
+  getRegionEditorErrorMessages,
+  getChrLocationStr
+} from '../browserHelper';
+import * as urlFor from 'src/shared/helpers/urlHelper';
+
+import applyIcon from 'static/img/shared/apply.svg';
+import clearIcon from 'static/img/shared/clear.svg';
+
+import styles from './BrowserRegionEditor.scss';
+import browserStyles from '../Browser.scss';
+import browserNavBarStyles from '../browser-nav/BrowserNavBar.scss';
+
+type BrowserRegionEditorProps = {
+  activeGenomeId: string | null;
+  chrLocation: ChrLocation;
+  genomeKaryotypes: GenomeKaryotype[];
+  regionEditorActive: boolean;
+  regionFieldActive: boolean;
+  regionValidationInfo: RegionValidationResponse | null;
+  regionValidationLoadingStatus: LoadingState;
+  changeBrowserLocation: (genomeId: string, chrLocation: ChrLocation) => void;
+  replace: Replace;
+  resetRegionValidation: () => void;
+  toggleRegionEditorActive: (regionEditorActive: boolean) => void;
+  validateRegion: (region: string) => void;
+};
+
+export const BrowserRegionEditor = (props: BrowserRegionEditorProps) => {
+  const [region, locationStart, locationEnd] = props.chrLocation;
+
+  const [regionInput, setRegionInput] = useState(region);
+  const [locationStartInput, setLocationStartInput] = useState(
+    getCommaSeparatedNumber(locationStart)
+  );
+  const [locationEndInput, setLocationEndInput] = useState(
+    getCommaSeparatedNumber(locationEnd)
+  );
+  const [isFormSubmitted, setIsFormSubmitted] = useState(false);
+  const [locationStartErrorMessage, setLocationStartErrorMessage] = useState<
+    string | null
+  >(null);
+  const [locationEndErrorMessage, setLocationEndErrorMessage] = useState<
+    string | null
+  >(null);
+
+  const getKaryotypeOptions = () =>
+    props.genomeKaryotypes.map(({ name }) => ({
+      value: name,
+      label: name,
+      isSelected: regionInput === name
+    }));
+
+  const updateRegionInput = (value: string) => {
```

Comment:
the word "region" is used in two senses within this file, which is confusing. In some places it means the whole combination of chromosome(or any other "stick")-start-end; but here it means the name of the chromosome (or any other "stick"). I suggest we change the name here — either to Chromosome, or (if we don't like to call chromosome what may turn out to be a plasmid or a scaffold, etc.) to Stick, or something. Or maybe change the Region part in RegionEditor to Location.

Reply:
I'll change it to `chrInput` for now.

Reply:
I changed this again to `stick` and the `regionInput` to be `stickInput` so it is a lot clearer.